lib/chef/platform/query_helpers.rb

```diff
@@ -36,6 +36,28 @@ def windows_server_2003?
         is_server_2003
       end
 
+      def windows_nano_server?
+        return false unless windows?
+        require 'win32/registry'
+
+        # This method may be called before ohai runs (e.g., it may be used to
+        # determine settings in config.rb). Chef::Win32::Registry.new uses
+        # node attributes to verify the machine architecture which aren't
+        # accessible before ohai runs.
+        nano = nil
+        key = "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Server\\ServerLevels"
```

Comment:
*facepalm*  I'm silly and misread the actual key you were reading out.  This makes perfect sense.